corehq/pillows/case_search.py

```diff
@@ -91,7 +91,8 @@ def _add_smart_types(dynamic_properties, domain, case_type):
     # We can't do that for properties like geo_points in ES v2, as `ignore_malformed` is broken
     gps_props = get_gps_properties(domain, case_type)
     gps_props.add(get_geo_case_property(domain))
-    _add_gps_smart_types(dynamic_properties, gps_props)
+    if len(gps_props):
```

Comment:
nit: this won't work with just `if gps_props`?

Reply:
I tried being explicit here so that it's clear to see that we're checking whether a set has elements. But I'm happy to shorten this if you feel the above suggestion makes the readability more clear.

Reply:
Sounds good, I have addressed in [1c22cb4](https://github.com/dimagi/commcare-hq/pull/35001/commits/1c22cb4e5d66583724cc63cadba8f0f9c83cf20a).

Reply:
Thinking through this again, there won't ever exist a scenario where the `gps_props` set is empty. This is because we call `get_geo_case_property()` which will always return a value.

Given the above, I'm going to remove the commits for the conditional since it will always pass.

Reply:
This PR has been rebased to remove commits for adding the conditional.